### DISCUSSION TITLE
Helix randomly starts in insert mode and delete first character

### DISCUSSION THREAD

**Discussion by Crispy13** [helix-editor/helix#10650] [Troubleshooting]
I built it from source. 

Helix randomly starts in insert mode and then the first character is removed.

**Comment by kirawi** [comment]
Please use the bug issue template. Are you able to reproduce with the release version? If so, are you sure that you don't have something else, such as your terminal, accidentally sending a sequence of characters to Helix on startup? Helix is hardcoded to start in normal mode in a way I don't think is possible to create the behavior you're experiencing.

**Reply by dead10ck** [reply]
Yeah this seems very likely to be a misbehaving terminal sending some sequence of characters at startup

**Comment by Crispy13** [comment]
Yes I built it with source tar in release.

I don't understand either why this error occur.
In my personal pc, the case does not occur but on the remote server.

I will try to inspect the issue myself first and close this if I get some evidences.

**Comment by Minemobs** [comment]
Hi, I also have this issue on one of my server, but it doesn't happen every time.
This issue only occur on one specific server, It never happened when I use helix on my other server or WSL

https://github.com/helix-editor/helix/assets/38861340/a265535f-e229-418b-aabb-2bd5b1845454

**Reply by kirawi** [reply]
Yeah, this looks like the key sequence to enter insert and delete is being sent to Helix based on the delay from when it starts to when it happens. I'd guess probably `c` is being sent since both the deletion and mode change happens in the same frame in the video. This is just speculation though, without being able to reproduce it I don't think this is easily debuggable.

**Reply by kirawi** [reply]
Actually, you can try rebinding `c` to something else to test.

**Comment by nitroxis** [comment]
I have this problem as well. Seems to happen randomly. Maybe it has something to do with Windows Terminal?
